tracing/tracer.go

```diff
@@ -17,26 +17,22 @@ var zipkinReporter reporter.Reporter
 var Tracer = opentracing.GlobalTracer()
 
 // InitTracer configure the global tracer
-func InitTracer(
-	serviceName, addr string,
-	sampleRate float64,
-	logger log.Wrapper,
-) {
+func InitTracer(serviceName, addr string, sampleRate float64,
+) error {
 	if addr == "" {
-		logger("Endpoint to send spans to is undefined")
-		return
+		return errors.New("Endpoint to send spans to is undefined")
 	}
 
 	zipkinReporter := zipkinhttp.NewReporter(addr)
 
 	endpoint, err := zipkin.NewEndpoint(serviceName, "0.0.0.0:0")
-	if err != nil && logger != nil {
-		logger(fmt.Sprintf("Unable to create local endpoint: %+v\n", err))
+	if err != nil {
+		return errors.Wrap(err, "Unable to create local endpoint")
```

Comment:
We are on go1.13, so we don't really need third party library (`github.com/pkg/errors`) for that :) We can just use `fmt.Errorf("Unable to create local endpoint: %w", err)` to wrap them.

Reply:
go1.13 error wrapping doesn't contain stack traces for where the original error occurred

Reply:
nit: Since those wrapped errors all have different prefixes, it would be easy to figure out which return it is, I don't think the call stack actually brings a lot of extra value here. If we really want the call stacks I'd prefer to pass in the logger which prints the call stack while logging, instead of wrap the call stack into the error (both because we need to use `%+v` to get them, and if the logging itself also prints the call stack the two call stacks will get mixed together).

But if you really prefer this one, I'm fine with it.